src/Notepads/Views/NotepadsMainPage.IO.cs

```diff
@@ -108,6 +108,30 @@ public async Task<int> OpenFiles(IReadOnlyList<IStorageItem> storageItems)
             return successCount;
         }
 
+        private async Task<StorageFile> OpenSaveAs(ITextEditor textEditor)
+        {
+            NotepadsCore.SwitchTo(textEditor);
+            StorageFile file = await FilePickerFactory.GetFileSavePicker(textEditor, true).PickSaveFileAsync();
+            NotepadsCore.FocusOnTextEditor(textEditor);
+            return file;
+        }
+
+        private async Task TrySave(ITextEditor textEditor, StorageFile file, bool rebuildOpenRecentItems)
+        {
+            try
+            {
+                await NotepadsCore.SaveContentToFileAndUpdateEditorState(textEditor, file);
+                var success = MRUService.TryAdd(file); // Remember recently used files
+                if (success && rebuildOpenRecentItems)
+                {
+                    await BuildOpenRecentButtonSubItems();
+                }
+            } catch (Exception)
+            {
+                throw new UnauthorizedAccessException("Access denied");
```

Comment:
There are other cases where you might get this exception. Like the drive could be removed or the cloud drive connection lost. You cannot assume it is access denied here.

Reply:
> There are other cases where you might get this exception. Like the drive could be removed or the cloud drive connection lost. You cannot assume it is access denied here.

I changed from UnauthorizedAccessException to IOException.

According to the documentation, the exception that is thrown when an I/O error occurs.

I think she fits the case. That is, if you are unable to save the file for any reason, open a Save As.

And when you use FileSavePicker and try to save to a location that is blocked by the system, windows automatically denies it stating the reason (if the location is blocked, it doesn't exist, real read only, etc).

![Untitled Project](https://user-images.githubusercontent.com/1365683/81465624-ab6c8700-91a1-11ea-9dce-1893f0925afe.gif)

Reply:
There are still few potential problems with this implementation. Here is mine based on yours: https://github.com/JasonStein/Notepads/pull/512/files
Can you take a look?

Reply:
Few problems I solved this my implementation:

1. It properly saves UTF-8-BOM or any encoding that has a preamble.
2. Call CompleteUpdatesAsync in case exception happens.
3. No self created exception anymore since I found that the exception type is actually "UnauthorizedAccessException" when file is read-only and you are trying to save.

Reply:
If that PR looks good to you, please do following:

1. Abandon this one and create a new one with my changes and send out a PR.
2. I will close mine and accept yours so you get the credits.
3. Before you do so, please help me test all scenarios again to make sure it works in all cases.

Another test case would be editing a file and save it while OneDrive/Dropbox is updating it. Not sure what will happen in that case but it should be handled correctly.

Reply:
I merged yours implementations, I performed the tests again and I fixed an problem when you tried save file not found or on storage removed, he showed the error dialog. Then I added "FileNotFoundException" on catch to solve it.

I performed tests in several scenarios, such as Dropbox while updating the file, removing external storage, deleting file in editing and deleting file in shared network edition.

Reply:
LGTM, thanks for the effort!